KebPOS/MainMenu.cs

```diff
@@ -42,10 +42,39 @@ internal void InitializeMenu()
                 case MainMenuSelections.ViewOrderDetails:
                     ViewOrderDetails();
                     break;
+                case MainMenuSelections.DeleteOrder:
+                    DeleteOrder();
+                    break;
             }
         }
     }
 
+    private void DeleteOrder()  // Buray� kodluyorsun
+    {
+        ViewOrders(_kebabController.GetOrders());
+        bool validId = false;
+        var allOrders = _kebabController.GetOrders();
+        int selectedOrderId = 0;
+
+        do
+        {
+            selectedOrderId = AnsiConsole.Ask<int>("Enter the Id of order to be deleted.");
+            validId = Validation.IsValidOrderId(selectedOrderId, allOrders);
+            if (!validId)
+                Console.WriteLine("Please enter a valid Id");
+
+        } while (!validId);
+
+        var areYouSure = AnsiConsole.Confirm($"[Red] This will delete the order[/] [yellow]#{selectedOrderId}[/][red] Are you SURE?[/]", false);
+        if (!areYouSure)
+        {
+            Console.Clear();
+            return;
+        }
+
+        var toBeDeleted = allOrders[selectedOrderId - 1];
```

Comment:
This won't work. Here you're picking the element by the order in the list, not by the Id. Use Linq's Single() to grab the item by the actual Id.

Reply:
Because the view orders menu shows the id in that way. İ cecked it. The id s shown does not match the ids of orders in the db. My method deletes the correct order shown on screen. İ checked it. If you can send me a test that dont work ı will appreciate it

Reply:
@Doc415 Yup my bad. I forgot the ids were being mapped. I'll merge this so you can merge the main branch to your other solution

Reply:
Thank you